carbon-design-system/carbon-addons-iot-react
# fix(svgs): convert to react components and update usage docs

Closes #1732 

**Summary**

- SVG files were not being converted to react components correctly, causing the external build to not export and render the svgs

**Change List (commits, features, bugs, etc)**

- Add `svgr/cli` which is a helper tool to convert SVG files into React component files (JSX)
- Added contributor documentation on how to add icons in the future
- Moved svg icons to a new folder `src/icons`
  - Kept the true svg image files in `src/icons/svg`
  - New JSX icon files in `src/icons/components`
- Updated CardGalleryList imports to use new JSX files
- Added a new eslint rule to disable imports of SVGs directly
- Added `svg-convert` yarn command to automatically convert any svg files in `src/icons/svg` into react component files in `src/icons/components`
- Added `svg-convert` command to `yarn build`

**Acceptance Test (how to verify the PR)**

- Go to a sandbox outside of our storybook production build
- Import DashboardEditor or CardEditor
- See that the icons are rendering correctly in the CardGalleryList on the righthand side

- Run `yarn build` and see that all SVG files are converted into react component files (you'll see the commands running in the terminal)
